src/components/DirectusAsset.tsx

```diff
@@ -5,26 +5,56 @@ import { DirectusContext } from '@/DirectusProvider';
 
 export const DirectusAsset = ({
   apiUrl: propsApiUrl,
+  accessToken: propsAccessToken,
   asset,
   download = false,
   render,
 }: DirectusAssetProps): JSX.Element => {
   const directusContext = React.useContext(DirectusContext);
+  const [assetUrl, setAssetUrl] = React.useState<string | undefined>();
```

Comment:
Question: because of the setAssetUrl call, are we adding an extra re-render here, right? Avoidable?

Reply:
Yes.

Maybe, the problem ist that `directus.auth.token` returns a promise, so we need to await it.
We could extract the whole generateUrl function and await the token in the useEffect and set the initial value of `assetUrl` to the value of the now syncrronous `generateUrl` function if the authState is `"unauthenticated"`.

But i think this would make the whole code more complicated and harder to read.

Reply:
On second thought, we could use the now intoduced useDirectusAuth hooks to store the token directly in the provider and switch back from the async `generateUrl` function to a simple useMemo.

I have to rebase this branch from the current main for that and add a couple more lines in multible files to this to work. Should i do that in this PR or shall we merge this and then I open a new PR to prevent the unnecessary renders?

Reply:
Do as you think it's more appropriate, thanks!